### PR TITLE
Docs: Update ColorPallete and example

### PR DESCRIPTION
**Pull request by aberllin**
## What I did

- Updated documentation to include support for gradients in ColorItem.
- Mentioned examples of valid gradient formats such as 'linear-gradient(to right, white, black)' and 'linear-gradient(65deg, white, black)'.

Related to the issue: [#10232](https://github.com/storybookjs/storybook/issues/10232)

### Documentation

- [x] Add or update documentation reflecting your changes

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: docs/api/doc-blocks/doc-block-colorpalette.mdx [hunk 1]
```diff
@@ -46,6 +46,18 @@ import { Meta, ColorPalette, ColorItem } from '@storybook/blocks';
...
+      subtitle="Grayscale"
+      colors={{
+        Gradient: 'linear-gradient(to right,white,black)',
+      }}
+  />
+ <ColorItem
+      title="gradient"
+      subtitle="Grayscale"
+      colors={['linear-gradient(65deg,white,black)']}
+  />
```
Comment (jonniebigodes) [inline docs/api/doc-blocks/doc-block-colorpalette.mdx:60]: @aberllin when you have a moment, can you adjust the spacing to match the rest of the example, as it's a bit off? It's using 4 spaces when it should be 2 spaces instead.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@aberllin, thanks for putting together this pull request and helping us improve the documentation by adding this small but important example. We appreciate it 🙏 ! 

I left a small item for you to look into when you can.

Let me know once you've addressed it, and I'll take a look. If nothing more stands out, I'll gladly merge this pull request.

I hope you have a fantastic day.

Stay safe

File: docs/api/doc-blocks/doc-block-colorpalette.mdx (hunk 1 above)
Comment (aberllin) [reply docs/api/doc-blocks/doc-block-colorpalette.mdx]: Hi @ jonniebigodes , please check again! 
Thank you

**Comment by jonniebigodes** [comment]
@aberllin thanks for addressing the feedback so promptly. Appreciate it 🙏 ! All is good on my end. Will merge once the checklist clears.

**Review by jonniebigodes — APPROVED** [review]

**Pull request merged**